src/test/resources/processor_test_files/2184_CastArithmeticOperand/NOCOMPILE_BinaryOperatorInMethodCallThatCanReceiveDifferentNumberOfParameters.java

```diff
@@ -0,0 +1,19 @@
+/*
+When a binary operator is in a method call that can receive different number of parameters (e.g., `void info(String var1, Object... var2);`),
+we can get an IndexOutOfBoundsException when trying to find the type expected for the argument (i.e., the binary operator in this case).
+This test checks that such an exception doesn't happen.
+*/
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class NOCOMPILE_BinaryOperatorInMethodCallThatCanReceiveDifferentNumberOfParameters {
+    float twoThirds = 2/3; // Noncompliant; THIS LINE WAS ONLY ADDED TO MAKE THE TEST FIND A VIOLATION IN THIS FILE SO THE NEXT LINES CAN BE ACTUALLY TESTED!
+
+    Logger LOGGER = LoggerFactory.getLogger(getClass());
+
+    void method() {
+        LOGGER.info("{} {}", 1, 1 + 1); // This is ok, see the method signature: void info(String var1, Object var2, Object var3);
+        LOGGER.info("{} {} {}", 1, 2, 1 + 2); // This can cause an IndexOutOfBoundsException, see the method signature: void info(String var1, Object... var2);
```

Comment:
Here, I think it would be better to define the methods you call directly in the test file. This is for three reasons.

1. `slf4j` is not actually a dependency of Sorald, it's a transitive dependency through Spoon and Sonar. If Spoon/Sonar were to remove it, this test case would suddenly fail as the type can't be resolved.
2. If `slf4j` changes either signature (it's moving to 2.0, so it could happen), the test case may break.
3. It would make the comments with the signatures redundant, as the declarations would be right there.

But this works now, so I think we can leave it for later.

Reply:
Done.